source/app.d

```diff
@@ -159,32 +228,51 @@ struct Comment { string url, body_; }
 Comment getBotComment(string commentsURL)
 {
     // the bot may post multiple comments (mention-bot & bugzilla links)
-    auto res = requestHTTP(commentsURL, (scope req) { req.headers["Authorization"] = githubAuth; })
+    auto res = ghGetRequest(commentsURL)
         .readJson[]
         .find!(c => c["user"]["login"] == "dlang-bot" && c["body"].get!string.canFind("Bugzilla"));
     if (res.length)
         return deserializeJson!Comment(res[0]);
     return Comment();
 }
 
-void ghSendRequest(T...)(HTTPMethod method, string url, T arg)
-    if (T.length <= 1)
+auto ghGetRequest(string url)
 {
+    return requestHTTP(url, (scope req) {
+        req.headers["Authorization"] = githubAuth;
+    });
+}
+
+
+void ghSendRequest(scope void delegate(scope HTTPClientRequest req) userReq, string url)
+{
+    HTTPMethod method;
     requestHTTP(url, (scope req) {
         req.headers["Authorization"] = githubAuth;
-        req.method = method;
-        static if (T.length)
-            req.writeJsonBody(arg);
+        userReq(req);
+        method = req.method;
     }, (scope res) {
         if (res.statusCode / 100 == 2)
-            logInfo("%s %s, %s\n", method, url, res.bodyReader.empty ?
-                    res.statusPhrase : res.readJson["html_url"].get!string);
+        {
+            logInfo("%s %s, %s\n", method, url, res.statusPhrase);
+            res.bodyReader.readAllUTF8;
+        }
         else
             logWarn("%s %s failed;  %s %s.\n%s", method, url,
                 res.statusPhrase, res.statusCode, res.bodyReader.readAllUTF8);
     });
 }
 
+auto ghSendRequest(T...)(HTTPMethod method, string url, T arg)
+    if (T.length <= 1)
+{
+    return ghSendRequest((scope req) {
+        req.method = method;
+        static if (T.length)
+            req.writeJsonBody(arg);
+    }, url);
+}
```

Comment:
The weird vibe.d request API is actually a good test case for the scope/DIP1000 topic. It's only using callbacks to not escape scoped classes.